src/database/index.tsx

```diff
@@ -379,6 +419,89 @@ export const saveUser = async (userData: UserToSave): Promise<void> => {
   }
 };
 
+export const addScenarioUser = async (
+  scenarioId: string,
+  email: string,
+): Promise<User | null> => {
+  const db = await getDb();
+
+  try {
+    const userResult = await db
+      .collection(usersCollectionId)
+      .where("email", "==", email)
+      .get();
+
+    if (userResult.docs.length > 1) {
+      throw new Error(`Multiple users found matching email ${email}`);
+    } else if (userResult.docs.length === 0) {
+      throw new Error(`No users found matching email ${email}`);
+    }
+
+    const userDocument = userResult.docs[0];
```

Comment:
since this is basically the same as what `getUser` is doing, I might just factor it out of `getUser` into an unexported function and just leave `getUser` to pass the result through `buildUser` (I am assuming that was the motivation here since you need the raw Firestore document)

Reply:
I agree that there's likely an opportunity to refactor some things and remove duplication in this file, but I'm not sure if I'm understanding what's being requested here.  The query and checks here are very similar to `getUser`, but we're looking up the user by `email` here since that's what will be passed in from the frontend (vs. `auth0Id` in `getUser`).  Is the suggestion to create an unexported function that is smart enough to deal with either having an `email` or `auth0Id` passed in as a parameter?

Reply:
oh whoops, I didn't even catch that. it's a little more involved than what I imagined when I originally commented but I would still say ... yes. (it could accept, e.g., an object with optional keys `email` and `auth0Id` and require that at least one of them has a value.) because the lookup logic is so finicky I think there is enough value in not having it duplicated that i still wouldn't punt on it

Reply:
Ok, cool.  I'll knock it out!